File: soma_v3.py
'Terceira versão da função em Python'

#função de adição
def soma (a,b):
    resultado = a + b
    return resultado

#captura o valor do teclado e converte para um numero inteiro 
a = int(input())

op = input()

if op == '+':
    
    b = int(input())
    #chamada da função
    resultado_somar = soma(a,b)

#mostra o resultado
print (resultado_somar)

File: soma_v3.cs
/******************************************************************************
terceira versão do programa soma
*******************************************************************************/
using System;
class Operacao {
  static void Main() {
    String op;
    int a;
    int b;
    //atribuição de dois valores as variaveis a. Int32.Parse converte para inteiro
    a = int.Parse(Console.ReadLine());//captura do teclado o valor digitado
    op = Console.ReadLine();
    
      switch (op)
    {
        case "+":
          b = int.Parse(Console.ReadLine());//captura do teclado o valor digitado
          //chamo a função e ao mesmo tempo passo as variaveis
          Console.WriteLine ("Resultado foi: "+soma(a,b));
          break;
        default:
          Console.WriteLine("Erro, operação não encontrada");
          break;
    }
  }
  
  static int soma (int a, int b){
      int resultado = a + b;
      return resultado;
  }
}